src/notation/view/internal/dynamicpopupmodel.h

```diff
@@ -0,0 +1,40 @@
+#ifndef MU_NOTATION_DYNAMICPOPUPMODEL_H
+#define MU_NOTATION_DYNAMICPOPUPMODEL_H
+
+#include <engraving/iengravingfontsprovider.h>
+#include <engraving/style/style.h>
+#include <view/abstractelementpopupmodel.h>
+
+#include <QList>
+#include <QObject>
+#include <QVariantList>
+
+using namespace mu::engraving;
+
+namespace mu::notation {
+class DynamicPopupModel : public AbstractElementPopupModel
+{
+    Q_OBJECT
+
+    INJECT(engraving::IEngravingFontsProvider, engravingFonts)
+
+    Q_PROPERTY(QVariantList pages READ pages NOTIFY pagesChanged)
+
+public:
+    explicit DynamicPopupModel(QObject* parent = nullptr);
+
+    Q_INVOKABLE void init() override;
+
+    QVariantList pages() const;
+
+signals:
+    void pagesChanged();
+
+private:
+    QList<QStringList> m_pages;
```

Comment:
From the name of this member, it is not very clear what it will contain (namely, the dynamics symbols as strings). It would be great if you could either adjust the name to make it more descriptive, or add a comment explaining it.